README.md

```diff
@@ -100,46 +96,108 @@ const useProduct = createRemoteResource({
+  const [products] = useResourceState(productsResource, [categoryId]);
+
+  return (
+    <section>
+      <header>
+        <h1>{categoriesById[categoryId].name}</h1>
+      </header>
+      <ul>
+        {products.map(product => (
+          <li key={product.id}>
+            <Link to={`/products/${product.id}`}>{product.name}</Link>
+          </li>
+        ))}
+      </ul>
+    </section>
+  );
+};
+```
+
+This hook is very powerful. Let's walk through what happens when it is used:
 
-// Bypasses the cache and calls `load` again but without throwing the promise. Returns the promise from `load`.
-actions.refresh();
+1. If there is no data in the store for this resource OR if there is no entry for the combination of arguments OR if the component is mounting and the last time the data was loaded for this entry was longer ago than the `invalidateAfter` argument then the `load` function that was passed into `createRemoteResource` will be invoked and the returned promise will be thrown until the promise resolves or rejects.
+2. If the promise resolves with the data then it will be available as the first item in the tuple.
+3. You can change the state using the second item in the tuple. Note that state changes are not local to the component like `useState`. If the component unmounts and remounts the state will be the same as when you left it.
 
-// Calls the `save` function if provided. Only calls `save` once as long as the promise is not resolved. Returns the promise from `save`.
-actions.remoteSave(updatedProduce);
+### `useResourceActions`
 
-// Calls the `delete` function if provided. Only calls `delete` once as long as the promise is not resolved. Returns the promise from `delete`.
-actions.remoteDelete();
+A React hook that takes a resource and an optional array of arguments and returns an object literal with the following methods:
 
-// Sets the value for this entry in the cache and stores the time it was updated. All components using the hook will be re-rendered with the new value. Can optionally take an updater function that receives the current value.
-actions.setCache({ ...product, tags: product.tags.concat("shoes") });
-actions.setCache(product => ({ ...product, tags: product.tags.concat("shoes") });
+- `set`: A function that takes either the new state to set for this entry or a function that takes the current state of the entry and returns what should be set as the new state for the entry.
+- `refresh`: A function that allows you to bypass the check against the `updatedAt` timestamp and immediately refetch the data. Note: the promise will **not** be thrown for this action.
+- `save`: The `save` function that was defined with `createRemoteResource`. Note: this will be `undefined` if you did not define a `save` function with `createRemoteResource`.
+- `delete`: The `delete` function that was defined with `createRemoteResource`. Note: this will be `undefined` if you did not define a `delete` function with `createRemoteResource`.
```

Comment:
```suggestion
- `delete`: The `delete` function that was defined with `createRemoteResource`. *Note: this will be `undefined` if you did not define a `delete` function with `createRemoteResource`.*
```